src/Lecture2.hs

```diff
@@ -70,7 +85,13 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt _ [] = (Nothing, [])
+removeAt index (x:xs)
+  | index == 0 = (Just x, xs)
+  | otherwise  = let (e , t) = removeAt (index - 1) xs
```

Comment:
One minor optimization is to return `Nothing` and the list itself immediately if the index is negative. Currently, your implementation hangs on negative indexes and infinite lists ♾️